controllers/pipelines/reconciler.go

```diff
@@ -32,15 +32,14 @@ func desiredProvider(resource pipelinesv1.HasProvider, config config.Configurati
 	return config.DefaultProvider
 }
 
-func loadProvider(ctx context.Context, reader client.Reader, namespace string, desiredProvider string) (pipelinesv1.Provider, error) {
+func (br ResourceReconciler[R]) loadProvider(ctx context.Context, namespace string, desiredProvider string) (pipelinesv1.Provider, error) {
```

Comment:
out of interest, why `br`?

Reply:
`base resource` I think, but its inline with all the others below